rfd/0157-aws-eks-discover.md

```diff
@@ -0,0 +1,225 @@
+to an EKS cluster (details [below](#eks-identity-provider-association)). AWS permissions config:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "eks:DescribeCluster",
+                "eks:ListClusters",
+                "eks:AssociateIdentityProviderConfig",
+                "eks:DisassociateIdentityProviderConfig",
+                "eks:ListIdentityProviderConfigs",
+                "eks:DescribeIdentityProviderConfig"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+We will reuse existing code for AWS Integration setup using these new permissions.
+
+After setting up AWS integration, users will see a list of the EKS clusters in the next step.
+There, they will be able to select the cluster they want to enroll.
+We will determine which clusters are already enrolled and present in the Teleport inventory; those clusters
+will be greyed out in the table.
+
+Users will also have an option to enable Kubernetes App Discovery in the enrolled clusters, this option
+will be enabled by default.
+
+Users will have two main options for enrolling EKS clusters:
+ - Enroll specific selected cluster from the list of available ones.
+ - Set up automatic discovery and enrollment of the clusters using OIDC integration.
+
+Cluster enrollment will be done through the installation of the Teleport agent using the Helm chart
+`teleport-kube-agent`.
+
+### Enrolling Selected Clusters
+
+For the enrollment of just the selected clusters, users will be further given two choices:
+- Use EKS OIDC identity provider association to enroll a cluster with just the API.
+- As a backup option - manually run a Helm command that we generated for them.
+
+For completely automated cluster enrollment through the API, the Helm Go SDK will be used -
+allowing us to avoid the need for an additional step to run Helm itself elsewhere. It would require
+running a job inside the target EKS cluster with a special image that has Helm on it, and this job
+would install the Teleport kube agent. By using the Helm Go SDK, we can run the installation directly from the Teleport
+process and don't need to maintain a new special image and EKS installation job.
+
+### Automatic Discovery and Enrollment
+
+Users will also be able to set up automatic discovery and enrollment of EKS clusters in the Discover UI.
+We will reuse the existing EKS discovery capability, although a new mode will be added to it - discovery through
+the AWS integration. EKS discovery in that mode will rely on the AWS integration, selected in the Discovery UI,
+to find clusters and then install the Helm chart on them. The algorithm will be as follows:
+- List EKS clusters through the AWS integration.
+- Cross-check to see which EKS clusters are already registered in the system, using labels we inject on Helm chart installation.
```

Comment:
Labels are not secret. Basically when we enrol EKS cluster we specify labels to the kube agent chart, that give information about this chart - aws region, cluster name, discovery source, etc (maybe cluster arn). So then when we list EKS clusters we can compare it to the list of Kube cluster we have already registered in the Teleport. So if, for example, we have EKS cluster "TestEKS" from "us-east-1" region and we also have Kube server in Teleport with labels saying same info, we know that we already enrolled this cluster.

Reply:
So in theory if you falsified a label to match another cluster already registered, it would just prevent a cluster from being registered with Teleport?  If you deleted a label and the cluster was registered again, would that be impactful in any way?

Reply:
> So in theory if you falsified a label to match another cluster already registered, it would just prevent a cluster from being registered with Teleport?

Yes, it would be disabled for enrolling.

> If you deleted a label and the cluster was registered again, would that be impactful in any way?

Not really, we'll check during enrolment if agent is already installed and abort enrolment if it is, but even if we don't check it would be just the same installation, so nothing really would be affected. We just use those labels to quickly filter already enrolled clusters, for better UX and so we don't have to go too deep on each one to see if it needs to be enrolled.

Reply:
Thank you.  We will need to continue to make sure that there is no possible sensitive interactions through label manipulation.  But given what you described above it seems like at most a denial of service would be possible, so I have no significant concerns right now.